modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,337 @@
+            parentClasses.add(enclosingElement.getSimpleName().toString());
+            enclosingElement = enclosingElement.getEnclosingElement();
+        }
+        StringBuilder className = new StringBuilder();
+        for (int index = parentClasses.size() - 1; index >= 0; index--) {
+            className.append(parentClasses.get(index));
+            className.append("$");
+        }
+        return className.toString() + classElement.getSimpleName();
+    }
+
+    private String getMethodIdentifierString(Element methodElement) {
+        TypeVisitor<ExecutableType, Void> executableTypeVisitor =
+            new SimpleTypeVisitor6<ExecutableType, Void>() {
+                @Override
+                public ExecutableType visitExecutable(
+                    ExecutableType executableType, Void obj
+                ) {
+                    return executableType;
+                }
+            };
+        ExecutableType method =
+            methodElement.asType().accept(executableTypeVisitor, null);
+        List<? extends TypeMirror> methodParams = method.getParameterTypes();
+
+        /*
+         * Fetching the return type of the method in the format ASM
+         * parses the return type in a method descriptor.
+         */
+        final String returnType;
+        returnType = getRepresentation(method.getReturnType());
+
+        List<String> methodArguments = new ArrayList<String>();
+
+        /*
+         * For a non-static inner class constructor, adding the outer class
+         * object as an argument.
+         * Example:
+         * fullClassName = "org.gaul.example_package.OuterClass$InnerClass"
+         * methodArguments += "org.gaul.example_package.OuterClass"
+         */
+        String fullClassName =
+            getClassHeader(methodElement.getEnclosingElement());
+        if (methodElement.getKind().equals(ElementKind.CONSTRUCTOR)) {
+            Set<Modifier> modifiers =
+                methodElement.getEnclosingElement().getModifiers();
+            if (!modifiers.contains(Modifier.STATIC)) {
+                int index = fullClassName.lastIndexOf("$");
+                if (index != -1) {
+                    methodArguments.add(fullClassName.substring(0, index));
+                }
+            }
+        }
+
+        methodArguments.addAll(getMethodArgumentTypes(methodParams));
+        return ModernizerAnnotationOutput.getMethodRep(
+            fullClassName, methodElement.getSimpleName().toString(),
+            returnType, methodArguments);
+    }
+
+    /**
+     * Fetches the method arguments in the format the way ASM parses a method
+     * descriptor.
+     * @param methodParams A list of types of formal parameters
+     * of an executable type
+     * @return A list of all the arguments after formatting
+     * Example:
+     * Input: {@code {int[][], String, List<Integer>}}
+     * Output: {@code {"int[][]", "String", "java.util.List"}}
+     */
+    private List<String> getMethodArgumentTypes(
```

Comment:
This could just be called `getRepresentations`. It's really not relevant to its functionality that it expects params